pulse_actions/authentication.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+USER = None
+PASSWORD = None
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+def get_user_and_password():
+    """
+    Retrieve user and password environment variables.
+
+    Raise AuthenticationError if either aren't defined.
+    """
+    global USER
+    global PASSWORD
+
+    if USER is None:
+        user = os.environ.get('PULSE_USER')
+    if PASSWORD is None:
+        password = os.environ.get('PULSE_PW')
+
+    if user is None and password is None:
+        raise AuthenticationError("PULSE_USER and PULSE_PW are not set.")
+    elif user is None:
+        raise AuthenticationError("PULSE_USER is not set.")
+    elif password is None:
+        raise AuthenticationError("PULSE_PW is not set.")
```

Comment:
Could we please still keep PULSE_ in the naming?
I believe we also have the TASKCLUSTER_ variables and I would still like to
keep the distinction.

I believe you can run flak8 on your changes to catch these kind of issues.
Perhaps we should file an issue to have tox.ini with calls to flake8 as we
do in Mozci.

Could you also please squash the changes before merging?
I wish GH would let us do that before merges.

On 17 September 2015 at 13:15, Vaibhav Agrawal <notifications@github.com>
wrote:

> In pulse_actions/authentication.py
> <https://github.com/adusca/pulse_actions/pull/18#discussion_r39772167>:
>
> > +    Raise AuthenticationError if either aren't defined.
> > +    """
> > +    global USER
> > +    global PASSWORD
> > +
> > +    if USER is None:
> > +        user = os.environ.get('PULSE_USER')
> > +    if PASSWORD is None:
> > +        password = os.environ.get('PULSE_PW')
> > +
> > +    if user is None and password is None:
> > +        raise AuthenticationError("PULSE_USER and PULSE_PW are not set.")
> > +    elif user is None:
> > +        raise AuthenticationError("PULSE_USER is not set.")
> > +    elif password is None:
> > +        raise AuthenticationError("PULSE_PW is not set.")
>
> Use capital USER and PASSWORD in the lines above
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adusca/pulse_actions/pull/18/files#r39772167>.
>



-- 
Zambrano Gasparnian, Armen
Automation & Tools Engineer
http://armenzg.blogspot.ca